template/app/src/androidTest/java/io/bloco/template/ui/counter/CounterActivityTest.kt

```diff
@@ -1,38 +1,67 @@
 package io.bloco.template.ui.counter
 
+import androidx.preference.PreferenceManager
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions.click
 import androidx.test.espresso.assertion.ViewAssertions.matches
 import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.espresso.matcher.ViewMatchers.withText
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import com.tfcporciuncula.flow.FlowSharedPreferences
+import dagger.hilt.android.testing.BindValue
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import io.bloco.template.AppModule
 import io.bloco.template.R
-import io.bloco.template.utils.ClearPreferencesTestRule
+import io.bloco.template.data.CounterRepository
+import kotlinx.coroutines.runBlocking
 import org.junit.Rule
 import org.junit.Test
+import org.junit.rules.RuleChain
 import org.junit.runner.RunWith
 
+@HiltAndroidTest
+@UninstallModules(AppModule::class)
 @RunWith(AndroidJUnit4::class)
 class CounterActivityTest {
 
-    @get:Rule
-    val clearPreferencesTestRule = ClearPreferencesTestRule()
+    @BindValue @JvmField val flowSharedPreferences : FlowSharedPreferences = let {
+        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(
+            getInstrumentation().targetContext)
+        sharedPreferences.edit().clear().commit()
+        val flowSharedPreferences =  FlowSharedPreferences(sharedPreferences)
 
-    @get:Rule
+        runBlocking {
+            flowSharedPreferences.getInt(CounterRepository.COUNTER_KEY, defaultValue = 1).setAndCommit(1)
+        }
+        flowSharedPreferences
+    }
```

Comment:
1) You already have a rule for clearing preferences. But you are also clearing them here?
2) Do you really need to replace with a new instance of FlowSharedPreferences? Can't you just inject here the existing FlowSharedPreferences instance provided, and just manipulate it? Or would that happen at the wrong time?

Reply:
3. You can also just inject the GetAndSetCounter and use it to set a value.

Reply:
about 1, I no longer have the rule I deleted it.
not sure if I can do 2, need to check it in more detail
3rd I can do, but the "idea" was to have hilt here as kinda of a basic example.

Reply:
You still need Hilt to inject `GetAndSetCounter` into the test ;)